java-client/session/src/main/java/io/deephaven/client/impl/FetchedObject.java

```diff
@@ -45,16 +52,42 @@ public void writeTo(OutputStream out) throws IOException {
         bytes.writeTo(out);
     }
 
+    /**
+     * The export ids.
+     *
+     * @return the export ids
+     * @deprecated use {@link #exports()}
+     */
+    @Deprecated
     public List<ExportId> exportIds() {
-        return exportIds;
+        return exports.stream().map(HasExportId::exportId).collect(Collectors.toList());
+    }
+
+    /**
+     * The exported server objects.
+     *
+     * @return the exported server objects
+     */
+    public List<ServerObject> exports() {
+        return exports;
     }
 
     @Override
     public String toString() {
         return "FetchedObject{" +
                 "type='" + type + '\'' +
                 ", bytes=" + bytes +
-                ", exportIds=" + exportIds +
+                ", exports=" + exports +
                 '}';
     }
+
+    /**
+     * Closes all of {@link #exports}.
```

Comment:
Why does close() release the exports, but not the object itself? How do you release the object itself - shouldn't FetchableObject.close() release the object's own ticket? 

Releasing the exports as the default operation doesn't really make sense to me - there's no reason that they shouldn't be able to outlive the object that first provided them.

Reply:
B/c there may or may _not_ be ownership of the original object. The interface exposes fetching via any arbitrary ticket; for example, when you fetch from the global scope, you don't close that object

```protobuf
message FetchObjectRequest {
    io.deephaven.proto.backplane.grpc.TypedTicket source_id = 1;
}
```

In the cases where you _are_ fetching from an export id, the caller _does_ have the control to manage the life of the object itself and the new typed_export_ids separately.

Reply:
If you fetch from the global scope without a re-export you're probably asking for trouble - consider having the API transparently (race a call to) reexport with the messagestream/fetch. Technically neither actually stops you from handing over a non-export ticket, but its a bad idea to use non-export tickets (and we document that at least).

I'm more scared of the "ServerObject.close() sometimes closes the object and leaves the children, and sometimes closes the children and leaves the object" - it doesn't look like it is going to encourage good code, but instead require that otherwise "generic" code typechecks what it is talking to before each close() call. Might as well make them different methods...

Reply:
ServerObject.close() never closes the children; it only ever closes the export associated w/ the ServerObject.